src/main/java/com/hillel/items_exchange/dao/CategoryRepository.java

```diff
@@ -13,4 +14,18 @@
     @Query(value = "SELECT name FROM category", nativeQuery = true)
     List<String> findAllCategoriesNames();
     Category findCategoryById(long id);
+    boolean existsByName(String name);
+    boolean existsByNameIgnoreCase(String name);
+    boolean existsByIdAndNameIgnoreCase(long categoryId, String categoryName);
+
+    @Query(value = "SELECT IF((SELECT COUNT(*) FROM category " +
+            "           WHERE (id = :categoryId AND upper(name) = :categoryName)) > 0" +
+            "               OR" +
+            "                   (SELECT EXISTS(SELECT * FROM category WHERE id = :categoryId)" +
+            "                       AND" +
+            "                   (SELECT NOT EXISTS(SELECT * FROM category WHERE upper(name) LIKE :categoryName))" +
+            "                       AND" +
+            "                   :categoryName <> ''), 'true', 'false')", nativeQuery = true)
+    boolean isExistsByIdAndNameOrNewCategoryNameHasNotDuplicate(@Param("categoryId") long categoryId,
```

Comment:
I'll think how to simplify such request...

Reply:
We don't need this method at all, I have just forgotten to remove it.